crates/tx/src/action.rs

```diff
@@ -27,6 +27,7 @@ pub enum Action {
     Gov(GovAction),
     Pgf(PgfAction),
     Masp(MaspAction),
+    IbcShielding,
```

Comment:
I was thinking that instead of `IbcShielding` perhaps we could just expand the `MaspAction` to be an enum like:

```
enum MaspAction {
    TxSection(Hash),
    InnerTxCmt(Hash)
}
```

The first case is the one we already have and its hash is the hash of the tx's `Masp` section. The second case is an alternative to `IbcShielding` and its hash is that of the `TxCommitments` (i.e. the inner tx hash). The ibc transaction could write the second variant when the tx is an incoming one. In the SDK then, we can look at the specific inner txs of the batch that contain a masp ibc tx and extract only their memos without looking at all of them.

This is a minor suggestion, feel free to ignore it

Reply:
Thanks, it looks better. However, the change for setting Hash into the action caused wasm compilation error on arm mac. Let's leave it as it is.